ckks/params.go

```diff
@@ -393,15 +374,15 @@ func (p Parameters) StandardParameters() (pckks Parameters, err error) {
 // ParametersLiteral returns the ParametersLiteral of the target Parameters.
 func (p Parameters) ParametersLiteral() (pLit ParametersLiteral) {
 	return ParametersLiteral{
-		LogN:         p.LogN(),
-		Q:            p.Q(),
-		P:            p.P(),
-		Pow2Base:     p.Pow2Base(),
-		Sigma:        p.Sigma(),
-		H:            p.HammingWeight(),
-		RingType:     p.RingType(),
-		DefaultScale: p.DefaultScale().Float64(),
-		LogSlots:     p.LogSlots(),
+		LogN:     p.LogN(),
+		Q:        p.Q(),
+		P:        p.P(),
+		Pow2Base: p.Pow2Base(),
+		Sigma:    p.Sigma(),
+		H:        p.HammingWeight(),
+		RingType: p.RingType(),
+		LogScale: int(math.Round(math.Log2(p.DefaultScale().Float64()))),
```

Comment:
LogScale should probably be an `int64`, not an `int`. Otherwise, this operation may overflow on 32-bit systems.

Reply:
By design, all parameters and variables in Lattigo that do not interact with polynomials coefficients (which are type `uint64`) use the default type which is `int`.

Reply:
OK. I suggest that we verify that `LogScale` doesn't overflow then by using a function:
```
fn panic_log2(value: float64) -> int {
  // Ensure `log2(value)` is smaller than 2^(32)-1
  assert(value < math.Exp2(float64((int64(1) << 32) - 1)))
  return math.Log2(value)
}
```

Reply:
I don't this this can ever happen because `2^{2^{32}} = 2^{4294967296}` isn't a number that can be represented, it will return `+Inf`. I have thus added a check in `rlwe.Scale` that will panic if the scaling factor is `Inf` (`NaN` value will by default trigger a panic if used to instantiate a `*big.Float`).

Reply:
`2^{2^32}` can be represented in `float64` ([see here](https://en.wikipedia.org/wiki/Double-precision_floating-point_format)).

Reply:
Actually no. I was wrong! Sorry!